src/components/alarm/AlarmController.h

```diff
@@ -29,42 +29,57 @@ namespace Pinetime {
   namespace Controllers {
     class AlarmController {
     public:
-      AlarmController(Controllers::DateTime& dateTimeController);
+      AlarmController(Controllers::DateTime& dateTimeController, Controllers::FS& fs);
 
       void Init(System::SystemTask* systemTask);
+      void SaveAlarm();
       void SetAlarmTime(uint8_t alarmHr, uint8_t alarmMin);
       void ScheduleAlarm();
       void DisableAlarm();
       void SetOffAlarmNow();
       uint32_t SecondsToAlarm() const;
       void StopAlerting();
-      enum class AlarmState { Not_Set, Set, Alerting };
       enum class RecurType { None, Daily, Weekdays };
       uint8_t Hours() const {
-        return hours;
+        return alarm.hours;
       }
       uint8_t Minutes() const {
-        return minutes;
+        return alarm.minutes;
       }
-      AlarmState State() const {
-        return state;
+      bool IsAlerting() const {
+        return isAlerting;
       }
-      RecurType Recurrence() const {
-        return recurrence;
+      bool IsEnabled() const {
+        return alarm.isEnabled;
       }
-      void SetRecurrence(RecurType recurType) {
-        recurrence = recurType;
+      RecurType Recurrence() const {
+        return alarm.recurrence;
       }
+      void SetRecurrence(RecurType recurrence);
 
     private:
+      // Versions 255 is reserved for now, so the version field can be made
+      // bigger, should it ever be needed.
+      static constexpr uint8_t alarmFormatVersion = 1;
+      struct AlarmData {
+        uint8_t version = alarmFormatVersion;
+        uint8_t hours = 7;
+        uint8_t minutes = 0;
+        RecurType recurrence = RecurType::None;
+        bool isEnabled = false;
+      };
+      bool isAlerting = false;
+      bool alarmChanged = false;
+
       Controllers::DateTime& dateTimeController;
+      Controllers::FS& fs;
       System::SystemTask* systemTask = nullptr;
       TimerHandle_t alarmTimer;
-      uint8_t hours = 7;
-      uint8_t minutes = 0;
+      AlarmData alarm;
       std::chrono::time_point<std::chrono::system_clock, std::chrono::nanoseconds> alarmTime;
-      AlarmState state = AlarmState::Not_Set;
-      RecurType recurrence = RecurType::None;
+
+      void LoadAlarmFromFile();
+      void SaveAlarmToFile() const;
```

Comment:
The naming of those methods can probably be improved : `Alarm` is recurrent with the name of the class, and `File` is an implementation detail that brings no meaningful information to the interface of the class. What would you think about `SaveSettings()` or even `Save()` and `Restore()` or `RestoreSettings()`?

Reply:
we've got the public function `SaveAlarm()`. The two mentioned are private functions.

I don't know how confusing it is to have `SaveSettings()` and `SaveAlarm()` :thinking: 

I think I like the `Save()/Restore()` pair the most and I think it is the least confusing alternative

Reply:
Yep, it looks better to me !

I noticed that `Alarm` is redundant in other methods and fields that existed before this PR, so no need to fix that in this PR. We might want to clean that in the future :)